tests/test-common.js

```diff
@@ -137,6 +140,33 @@ describe('JSON-LD Signatures', () => {
     });
   });
 
+  context('PublicKeyProofPurpose', async () => {
```

Comment:
The `PublicKeyProofPurpose` class is for backwards compatibility/deprecated proofs. We should at least test this with `AssertionProofPurpose` because it's a commonly used proof purpose. We should be able to just move it to where other tests of that purpose are then.

The verification method should also have the security v2 context on it (can be copied from the proof below).

Reply:
> We should be able to just move it to where other tests of that purpose are then.
> 
> The verification method should also have the security v2 context on it (can be copied from the proof below).

This is where I ran into the issue. The AssertionProofPurpose block is hardwired to a key with an owner. We actually don't test any keys in any of the suites with `controller` or with `controller` and an object. Perhaps I can expand the keys to test for those scenarios?

Reply:
@aljones15, if by "expand the keys" you mean add more and run more tests (vs. change existing mock data/tests), then +1 to that. So -1 to changing any existing stuff, +1 to being additive.

Reply:
@dlongley  ok so basically I can copy the existing `sign and verify` test for `AssertionProofPurpose`, remove the `controller` being passed into the purpose, change the signer's id to point at a key with a controller object, and hope for the best. I think.

Reply:
Ok this is now done: https://github.com/digitalbazaar/jsonld-signatures/pull/91/commits/23a27c2a76f29aa59ed51b36bcc4ef7a65d61a29

and also here:

https://github.com/digitalbazaar/jsonld-signatures/pull/91/commits/6e33e9218dfd400e65e59eb1240eb00be1a78ba5

one weird thing I did notice: despite having public keys with:
```
controller: {id: 'foo'}
```

when it reaches the verificationMethod it has been reduced to just `controller: foo` so I'm not sure which Purpose @gjgd was using.